storage/mysql/storage_test.go

```diff
@@ -293,68 +228,59 @@ func diffNodes(got, want []storage.Node) ([]storage.Node, []storage.Node) {
 	return missing, extra
 }
 
-// cleanTestDB deletes all the entries in the database. Only use this with a test database
-// TODO(gdbelvin): Migrate to testonly/integration
+func openTestDBOrDie() *sql.DB {
+	db, err := OpenDB("test:zaphod@tcp(127.0.0.1:3306)/test")
+	if err != nil {
+		panic(err)
+	}
+	return db
+}
+
+// cleanTestDB deletes all the entries in the database.
 func cleanTestDB(db *sql.DB) {
 	// Wipe out anything that was there for this log id
 	for _, table := range allTables {
-		_, err := db.Exec(fmt.Sprintf("DELETE FROM %s", table))
-
-		if err != nil {
+		if _, err := db.Exec(fmt.Sprintf("DELETE FROM %s", table)); err != nil {
 			panic(fmt.Errorf("Failed to delete rows in %s: %s", table, err))
 		}
 	}
 }
 
-// TODO(codingllama): Migrate to Admin API
-func createTestDB(db *sql.DB) {
-	_, err := db.Exec(`REPLACE INTO Trees(TreeId, KeyId, TreeType, LeafHasherType, TreeHasherType)
-					 VALUES(23, "hi", "LOG", "SHA256", "SHA256")`)
+// createMapForTests creates a map-type tree for tests. Returns the treeID of the new tree.
+func createMapForTests(db *sql.DB) int64 {
+	tree, err := createTree(db, storageto.MapTree)
 	if err != nil {
-		panic(err)
-	}
-}
-
-// prepareTestTreeDB removes all database contents for the specified log id so tests run in a predictable environment. For obvious reasons this should only be allowed to run against test databases. This method panics if any of the deletions fails to make sure tests can't inadvertently succeed.
-// TODO(gdbelvin): Migrate to testonly/integration / create a new DB for freshness
-func prepareTestTreeDB(db *sql.DB, treeID int64, t *testing.T) {
-	// Wipe out anything that was there for this tree id
-	for _, table := range allTables {
-		_, err := db.Exec(fmt.Sprintf("DELETE FROM %s WHERE TreeId=?", table), treeID)
-
-		if err != nil {
-			t.Fatalf("Failed to delete rows in %s for %d: %s", table, treeID, err)
-		}
+		panic(fmt.Sprintf("Error creating map: %v", err))
```

Comment:
Discussed offline: we'll keep the panics for now. The "proper" solution would be to return the error; that wouldn't be a problem if our tests were more table-driven, but alas, they aren't (at least the ones in question and at least for now).